src/Transport/Rest/JsonStreamDecoder.php

```diff
@@ -83,10 +85,30 @@ public function __construct(StreamInterface $stream, $decodeType, $options = [])
      * completes. Throws an Exception if the stream is closed before the closing
      * byte is read or if it encounters an error while decoding a message.
      *
-     * @throws Exception
+     * @throws RuntimeException
      * @return \Generator
      */
     public function decode()
+    {
+        try {
+            foreach ($this->_decode() as $response) {
+                yield $response;
+            }
+        } catch (RuntimeException $re) {
+            $msg = $re->getMessage();
+            $streamClosedException =
+                strpos($msg, 'Stream is detached') ||
+                strpos($msg, 'Unexpected stream close');
```

Comment:
A caviat to `strpos` is it's recommended to use `false ===`, because otherwise you could have a situation where the string starts with one of these values (such as `"Stream is detached"`) and in that case, it would be equivalent to false because `0` evaluates as falsey in PHP.

Is that the expected behavior here?

Reply:
We want to test if exception message contains (or starts with) either of these messages. I will do `!== false` in case we get `0`. Thanks!

Reply:
I definitely think returning an exception for a status is a bad idea. I think what you're looking for is [Google\Rpc\Status](https://github.com/googleapis/common-protos-php/blob/master/src/Rpc/Status.php)